plugins/package-managers/pub/src/main/kotlin/Pub.kt

```diff
@@ -597,6 +599,17 @@ class Pub(
                         logger.warn { "No version information found for package $rawName." }
                     }
 
+                    val hostUrl = pkgInfoFromLockFile["description"]["url"].textValueOrEmpty()
+
+                    val sourceArtifact = if (source == "hosted" && hostUrl.isNotEmpty() && version.isNotEmpty()) {
+                        val sha256 = pkgInfoFromLockFile["description"]["sha256"].textValueOrEmpty()
+                        val hash = if (sha256.isNotEmpty()) Hash(sha256, HashAlgorithm.SHA256) else Hash.NONE
```

Comment:
YOu should be able to just use `Hash.create()`. I just realized that its docs are slightly wrong, as for an empty string `Hash.NONE` is returned instead of `Hash.UNKNOWN`.

Reply:
I'll make a docs fix for that.

Reply:
See https://github.com/oss-review-toolkit/ort/pull/8179.